client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -54,47 +57,52 @@ class DeployedModel:
         # <DeployedModel 01234567-0123-0123-0123-012345678901>
 
     """
-    def __init__(self, _host=None, _run_id=None, _from_url=False, **kwargs):
-        # this is to temporarily maintain compatibility with anyone passing in `socket` and `model_id` as kwargs
-        # TODO: instate `host` and `run_id` params
-        # TODO: remove the following block of param checks
+    def __init__(self, host=None, run_id=None, creds=None, access_token=None, prediction_url=None, **kwargs):
         # TODO: put automodule verta.deployment back on ReadTheDocs
-        if 'socket' in kwargs:
-            warnings.warn("`socket` will be renamed to `host` in an upcoming version",
-                          category=FutureWarning)
-        if 'model_id' in kwargs:
-            warnings.warn("`model_id` will be renamed to `run_id` in an upcoming version",
-                          category=FutureWarning)
-        host = kwargs.get('host', kwargs.get('socket', _host))
-        run_id = kwargs.get('run_id', kwargs.get('model_id', _run_id))
-        if host is None:
-            raise TypeError("missing required argument: `host`")
-        if run_id is None:
-            raise TypeError("missing required argument: `run_id`")
-
-        self._session = requests.Session()
-        if not _from_url:
-            self._session.headers.update({_utils._GRPC_PREFIX+'source': "PythonClient"})
-            try:
-                self._session.headers.update({_utils._GRPC_PREFIX+'email': os.environ['VERTA_EMAIL']})
-            except KeyError:
-                six.raise_from(EnvironmentError("${} not found in environment".format('VERTA_EMAIL')), None)
-            try:
-                self._session.headers.update({
-                    _utils._GRPC_PREFIX+'developer_key': os.environ['VERTA_DEV_KEY'],
-                    _utils._GRPC_PREFIX+'developer-key': os.environ['VERTA_DEV_KEY'],  # see Client.__init__()
-                })
-            except KeyError:
-                six.raise_from(EnvironmentError("${} not found in environment".format('VERTA_DEV_KEY')), None)
+        creds = creds or credentials.load_from_os_env()
+        self._session = self._make_session(creds, access_token)
+
+        # TODO: remove the following block of param checks
+        # this is to temporarily maintain compatibility with anyone passing in `socket` and `model_id` as kwargs
+        self._host = self._validate_host(host, kwargs.get('socket'))
+        self._id = self._validate_id(run_id, kwargs.get('model_id'))
```

Comment:
(as mentioned in our earlier conversation) I'd be alright with ripping this all out and supporting the `from_url()` flow. It's unlikely that users are still instantiating this using a run/model ID.

Reply:
This actually leads to a substantial simplification of this class which I'm very in favor of.